mirrord-agent/src/steal.rs

```diff
@@ -0,0 +1,374 @@
+use std::{
+    collections::{HashMap, HashSet},
+    net::SocketAddr,
+    path::PathBuf,
+};
+
+use mirrord_protocol::{
+    tcp::{DaemonTcp, LayerStealTcp, NewTcpConnection, TcpClose, TcpData},
+    ConnectionID, Port,
+};
+use rand::distributions::{Alphanumeric, DistString};
+use streammap_ext::StreamMap;
+use tokio::{
+    io::{AsyncWriteExt, ReadHalf, WriteHalf},
+    net::{TcpListener, TcpStream},
+    select,
+    sync::mpsc::{Receiver, Sender},
+};
+use tokio_stream::StreamExt;
+use tokio_util::io::ReaderStream;
+use tracing::{debug, error, info, log::warn};
+
+use crate::{
+    error::{AgentError, Result},
+    runtime::set_namespace,
+};
+
+struct SafeIpTables {
+    inner: iptables::IPTables,
+    chain_name: String,
+}
+
+const IPTABLES_TABLE_NAME: &str = "nat";
+
+impl SafeIpTables {
+    pub fn new() -> Result<Self> {
```

Comment:
Add some docs here specifying which flags/configuration are being used for iptables (no need to say what they do, I think it's enough help just so that no one has to read the function implementation just to get this info).